parquet-hadoop/src/main/java/parquet/hadoop/ParquetInputFormat.java

```diff
@@ -253,7 +283,23 @@ public ParquetInputFormat() {
   @Override
   public List<InputSplit> getSplits(JobContext jobContext) throws IOException {
     Configuration configuration = ContextUtil.getConfiguration(jobContext);
-    return new ArrayList<InputSplit>(getSplits(configuration, getFooters(jobContext)));
+    List<InputSplit> splits = new ArrayList<InputSplit>();
+
+    if (isTaskSideMetaData(configuration)) {
```

Comment:
I chose to do this because I think the eventual goal is to remove the client-side strategy. In that case, we would remove the `isTaskSideMetaData` check and the `getSplits` code will be just this method, without the `getSplits(conf, footers)`. As you note, there isn't much sense in just one split strategy so that would be removed as well.

But, the easiest way to get this commit done was to keep the split strategy for now so that I didn't have to change much in that code and we can be confident it works as it did before this commit. I saw no reason to move it all back into the static space of `ParquetInputFormat` and get rid of the strategy abstraction before we could get rid of the last remaining strategy.

I think this is the cleanest way to get to where we can eventually remove the strategy entirely. But, I should definitely deprecate methods under this one that will be removed.